### PR TITLE
node: cleanup pex initialization

### PR DESCRIPTION
**Pull request by tychoish**
this doesn't fix the e2e tests along either of the dimensions that I'm
seeing at the moment (e.g. excessive ram use, and test
failure/timeout,) but it removes some clearly incorrect behabior which
makes the logs much more clear.

### CODE REVIEW

File: p2p/router.go [hunk 1]
```diff
@@ -176,14 +175,14 @@ func (o *RouterOptions) Validate() error {
...
-		o.IncomingConnectionWindow = 100 * time.Millisecond
+		o.IncomingConnectionWindow = time.Second
 	case o.IncomingConnectionWindow < time.Millisecond:
 		return fmt.Errorf("incomming connection window must be grater than 1m [%s]",
 			o.IncomingConnectionWindow)
 	}
 
 	if o.MaxIncomingConnectionAttempts == 0 {
-		o.MaxIncomingConnectionAttempts = 100
+		o.MaxIncomingConnectionAttempts = 10000
```
[the remark this answers is not in the document]
Comment (tychoish) [reply p2p/router.go]: oh, I missed reverting this. thanks for the catch.

 I was getting a lot of connection churn

File: test/e2e/networks/simple.toml [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+
 [node.validator01]
+  disable_legacy_p2p = true
 [node.validator02]
+  disable_legacy_p2p = true
 [node.validator03]
+  disable_legacy_p2p = true
 [node.validator04]
-
+  disable_legacy_p2p = true
```
Comment (cmwaters) [inline test/e2e/networks/simple.toml:16]: Maybe revert this back to what it was. I guess at some point we will make the transition to defaulting with the new p2p stack

File: node/node.go [hunk 3]
```diff
@@ -348,27 +348,6 @@ func NewNode(config *cfg.Config,
...
-	err = sw.AddUnconditionalPeerIDs(strings.SplitAndTrimEmpty(config.P2P.UnconditionalPeerIDs, ",", " "))
-	if err != nil {
-		return nil, fmt.Errorf("could not add peer ids from unconditional_peer_ids field: %w", err)
-	}
-
-	addrBook, err := createAddrBookAndSetOnSwitch(config, sw, p2pLogger, nodeKey)
-	if err != nil {
-		return nil, fmt.Errorf("could not create addrbook: %w", err)
-	}
-
```
Comment (cmwaters) [inline node/node.go:371]: Good work here. I forgot that with my changes to the p2p channels we were no longer reliant on setting up the switch when using the new p2p stack. Hopefully there's no other parts of the code that depend on switch being non nil

**Review by cmwaters — APPROVED** [review]
🧹 

(I would however call this more a change of node than e2e)

**Review by alexanderbez — APPROVED** [review]

File: node/node.go (hunk 3 above)
Comment (tychoish) [reply node/node.go:371]: hopefully indeed! 

if there are it'd be better to panic so we can fix it :)

**Pull request merged**